From ec2ca7e505e6bfe1fff2a7d6e6bb39b60b93e19a Mon Sep 17 00:00:00 2001
From: Haiqing Bai <Haiqing.Bai@windriver.com>
Date: Fri, 9 Dec 2016 15:23:17 +0800
Subject: [PATCH] nettle: check header files of openssl only if
 'enable_openssl=yes'.

The original configure script checks openssl header files to generate
config.h even if 'enable_openssl' is not set to yes, this made inconsistent
building for nettle.

Upstream-Status: Pending
Signed-off-by: Haiqing Bai <Haiqing.Bai@windriver.com>

refactored for 3.4. pending not in as of 3.4

Signed-off-by: Armin Kuster <akuster@mvista.com>

Refactor for 3.10.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>
---
 configure.ac | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7c003bb..779f7ff 100644
--- a/configure.ac
+++ b/configure.ac
@@ -197,9 +197,11 @@ AC_TYPE_SIZE_T
 AC_CHECK_SIZEOF(long)
 AC_CHECK_SIZEOF(size_t)
 
-AC_CHECK_HEADERS([openssl/evp.h openssl/ec.h openssl/rsa.h],,
-[enable_openssl=no
- break])
+if test "x$enable_openssl" = "xyes"; then
+  AC_CHECK_HEADERS([openssl/evp.h openssl/ec.h openssl/rsa.h],,
+  [enable_openssl=no
+   break])
+fi
 
 # For use by the testsuite
 AC_CHECK_HEADERS([valgrind/memcheck.h])
